src/library-authoring/data/apiHooks.ts

```diff
@@ -722,3 +723,28 @@ export const useUpdateContainerChildren = (containerId?: string) => {
     },
   });
 };
+
+/**
+ * Remove components from container
+ */
+export const useRemoveContainerChildren = (containerId?: string) => {
+  const queryClient = useQueryClient();
+  return useMutation({
+    mutationFn: async (usageKeys: string[]) => {
+      if (!containerId) {
+        return undefined;
+      }
+      return removeLibraryContainerChildren(containerId, usageKeys);
+    },
+    onSettled: () => {
+      if (!containerId) {
+        return;
+      }
+      // NOTE: We invalidate the library query here because we need to update the library's
+      // container list.
```

Comment:
What is the library's container list? We can't invalidate just that part?

Reply:
It is necessary to invalidate the query of the list of components/containers (from the search index) of the library, to update the component counter of the affected container. I will update the coment